sos/plugins/__init__.py

```diff
@@ -517,6 +517,9 @@ def add_copy_spec(self, copyspecs, sizelimit=None, tailit=True):
         file in a glob is too large it will be tailed to meet the sizelimit.
         """
 
+        if self.get_option('all_logs'):
+            sizelimit = None
+
```

Comment:
I don't necessarily disagree with this change, and it potentially simplifies the use of `all_logs` and `log_size` for plugin authors, BUT, there's a big chunk of this patch missing: if we are going to automate this inside of `add_copy_spec()` then there are a _tonne_ of locations in plugin code that need updating to modify or remove `all_logs` logic handling, e.g.:

```python
        self.limit = (None if self.get_option("all_logs")
                      else self.get_option("log_size"))
```

Following this change plugins do not need to modify the size limit themselves when `all_logs` is in effect: that's a good thing, but there's a whole lot of effectively dead code that needs to be reviewed and removed following this change.